packages/dds/map/src/directory.ts

```diff
@@ -693,17 +698,34 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         return this.localValueMaker.fromSerializable(serializable);
     }
 
+    private getParentDirectory(relativePath: string): IDirectory | undefined {
+        const absolutePath = this.makeAbsolute(relativePath);
+        const subdirs = absolutePath.substr(1).split(posix.sep);
+        if (subdirs.length === 0) {
+            return undefined;
+        }
+        subdirs.pop();
+        return this.getWorkingDirectory(subdirs.join(posix.sep));
+    }
+
+    private isSubDirectoryDeletePending(relativePath: string): boolean {
+        const parentSubDir = this.getParentDirectory(relativePath) as SubDirectory | undefined;
+        const index = relativePath.lastIndexOf(posix.sep);
+        const dirName = relativePath.substring(index + 1);
```

Comment:
can this return -1 (no slashes)? Do we get correct result in such case?

Reply:
Yes, even if it returns -1 we will get correct result because then dirName would be equal to relativePath.